src/Yoti/Util/Age/BaseAgeProcessor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Yoti\Util\Age;
+
+use Yoti\Entity\Attribute;
+
+class BaseAgeProcessor implements AgeProcessorInterface
+{
+    // You must define this value in the child class
+    const AGE_DELIMITER = '';
+
+    // You must define this pattern in the child class
+    const AGE_RULE_PATTERN = '';
+
+    /**
+     * @var Attribute
+     */
+    protected $attribute;
+
+    /**
+     * BaseAgeProcessor constructor.
+     *
+     * @param Attribute $attribute
+     */
+    public function __construct(Attribute $attribute)
+    {
+        $this->attribute = $attribute;
+    }
+
+    /**
+     * Return true if attribute name matches the pattern.
+     *
+     * @param Attribute $attribute
+     *
+     * @return bool
+     */
+    protected function isDerivedAttribute(Attribute $attribute)
+    {
+        return preg_match(static::AGE_RULE_PATTERN, $attribute->getName()) ? true : false;
+    }
+
+    /**
+     * This method could be overridden by a child class.
+     * Depending on the parsing process and complexity.
+     *
+     * @param Attribute $attribute
+     *
+     * @return array
+     */
+    protected function extractAgeVerificationData(Attribute $attribute)
+    {
+        $ageCheckArr = explode(static::AGE_DELIMITER, $attribute->getName());
+        if (count($ageCheckArr) > 1) {
+            return [
```

Comment:
Why not just get this object to return an AgeVerification object?
Looks like everything we need is here already.
At present we return a map here.  Then in _Processor_ we use that map and the original Attribute to create an AgeVerification.  Why not just create the AgeVerification straight away, and save the hassle?

Reply:
As you wish!